src/clarity-angular/layout/vertical-nav/providers/vertical-nav.service.ts

```diff
@@ -0,0 +1,92 @@
+import {Injectable} from "@angular/core";
+import {Subject} from "rxjs/Subject";
+import {Observable} from "rxjs/Observable";
+import {VerticalNavGroup} from "../vertical-nav-group";
+/**
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+@Injectable()
+export class VerticalNavService {
+    private _animateOnCollapsed: Subject<boolean> = new Subject<boolean>();
+
+    get animateOnCollapsed(): Observable<boolean> {
+        return this._animateOnCollapsed.asObservable();
+    }
+
+    private _collapsedChanged: Subject<boolean> = new Subject<boolean>();
+
+    get collapsedChanged(): Observable<boolean> {
+        return this._collapsedChanged.asObservable();
+    }
+
+    private _collapsed: boolean = false;
+
+    get collapsed(): boolean {
+        return this._collapsed;
+    }
+
+    set collapsed(value: boolean) {
+        value = !!value;
+        if (this.collapsible && (this._collapsed !== value)) {
+            this._animateOnCollapsed.next(value);
+            this._collapsed = value;
+            this._collapsedChanged.next(value);
+        } else if (!this.collapsible && this._collapsed) {
```

Comment:
i should have commented on this but this is to handle an edge case.

Edge case:
1. User has the nav marked as collapsible.
2. User collapses the nav
3. In collapsed mode the user marks the nav as not collapsible but the nav is still collapsed

So I thought by default this should set the nav as collapsed.

But now that i wrote the whole description, I should have just checked when the nav is being marked as not collapsible, make sure the nav is not collapsed. So i will change this logic to do that. Thanks for raising an issue on this.